### DISCUSSION TITLE
Add support for gfm alerts

### DISCUSSION THREAD

**Discussion by davidejones** [yuin/goldmark#422] [Ideas]
This issue is to propose that goldmark Github Flavored Markdown supports alerts which are documented here 
https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

Github Markdown is written as follows...

```
> [!NOTE]
> Highlights information that users should take into account, even when skimming.

> [!IMPORTANT]
> Crucial information necessary for users to succeed.

> [!WARNING]
> Critical content demanding immediate user attention due to potential risks.
```

Which renders in github as...

> [!NOTE]
> Highlights information that users should take into account, even when skimming.

> [!IMPORTANT]
> Crucial information necessary for users to succeed.

> [!WARNING]
> Critical content demanding immediate user attention due to potential risks.

**Comment by yuin** [comment]
I want to clarify my thought about this kind of feature requests.

I wont implement features (into goldmark repository) that is not defined in [Official Github Flavored Markdown spec](https://github.github.com/gfm/).

goldmark is a CommonMark parser. CommonMark aims to resolve compatibilities problems between Markdown parsers. Many markdown parser authors implement variety of markups by loosely defined document(like the URL you pasted). It leads markdown compatibility hells.

Many markdown users think like "emphasis in markdown is just put surrounding `*` to the words" . 
Then, how do you think should `**aaa* bb***` be parsed? Prior to CommonMark, markdown parser authors implemented as they think.
It leads markdown compatibility hells. CommonMark has 17 rules for emphasis <https://spec.commonmark.org/0.30/#emphasis-and-strong-emphasis>.

I wont implement features (into goldmark repository) that is not defined in [Official Github Flavored Markdown spec](https://github.github.com/gfm/) to avoid compatibilities hell.

If you really need to this markup, you have choices:

1. Make a request to Github to adopt the alerts markup as  [Official Github Flavored Markdown spec](https://github.github.com/gfm/)
2. Implement the alerts markup as a third-party goldmark extension

**Reply by davidejones** [reply]
Thanks @yuin! It's understandable you are sticking to the spec, i didn't realise github documented non spec features like this for use.

I would attempt to write a third party goldmark extension however this is for use in Hugo and I don't believe hugo accepts third party extension usage yet so it adds another layer to my problem.

For now i'll see if i can find some way to request github include this in the spec 🤔

**Reply by jmooring** [reply]
@davidejones You might consider using [markdown attributes](https://discourse.gohugo.io/t/markdown-attributes/41783).

**Reply by davidejones** [reply]
Interesting thanks @jmooring ! this change must have passed me by. This seems like a potential solution here, i'll try it out.

In the meantime i'm still not sure where to request/suggest spec changes with github 🤯

**Reply by yuin** [reply]
@davidejones 

>  i didn't realise github documented non spec features like this for use.

 Referring to https://github.com/github/cmark-gfm/issues/350 may be helpful to understand this question.
It seems Github does not implement alerts in markdown parser layer(≒goldmark layer). Github implements some markups by outer layer of markdown parsers(i.e. JavaScript DOM manupulation).
They probably wont write a official spec for this kind of markups(partially understandably).
